### DISCUSSION TITLE
List of error codes marked as fatal

### DISCUSSION THREAD

**Discussion by niro313A** [edenhill/librdkafka#3738] [General]
Is there any documentation which includes list of error codes which are marked as fatal errors?

**Comment by edenhill** [comment]
The severity of the error code depends on the context / API, so there's unfortunately no generic way to know this.
New APIs typically use the richer `rd_kafka_error_t` error type which has attributes to indicate if the error is temporary, fatal, etc. 
We're looking into ways to convey this information for existing/legacy APIs.

**Reply by niro313A** [reply]
In a scenario where idempotent producer is used to write messages to kafka cluster and event_cb is used to capture any fatal errors, application can check whether the error is fatal and trigger kafka client re-init or any other Fault Tolerant mechanism as required.
(event type will be RdKafka::Event::EVENT_ERROR and event marked as fatal_=true)

What type of errors are marked as fatal when event_cb is triggered?
(For e.g. disconnection from brokers etc?)

**Reply by edenhill** [reply]
No, only errors that would jeopardize the idempotent producer's guarantees, which are ever only seen when the cluster is in a very bad state.

The client will attempt to auto recover from all other errors.
